packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -1514,6 +1514,24 @@ export default class RestApi {
         res.sendStatus(200)
       })
     )
+
+    api.delete(
+      '/:collection(groups)/:id',
+      wrap(async (req, res) => {
+        const { id } = req.params
+        try {
+          await app.deleteGroup(id)
+          res.sendStatus(204)
+        } catch (error) {
+          json({ error: error.message })
```

Comment:
`json()` is a middleware.
If you want to prepare a `json` response, you have to call `json()` from the response. `res.json(...)`